src/lib.rs

```diff
@@ -0,0 +1,23 @@
+use thiserror::Error;
+
+mod big_int;
+mod macros;
+
+pub use crate::big_int::BigInt;
+pub use crate::big_int::RNSRepresentation;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum CrateError {
+    #[error("Error while converting!")]
+    ConversionError,
+    #[error("Invalid size, expected {expected:?}, found {found:?}")]
+    SizeError { expected: usize, found: usize },
+    #[error("Cannot divide by 0!")]
+    DivisionByZero,
+    #[error("Cannot divide {dividand:?} by {divisor:?}")]
+    DivisionError { dividand: BigInt, divisor: BigInt },
+    #[error("Modulus cannot be an even number!")]
+    EvenModulus,
+    #[error("Inverse of {operand:?} does not exist given modulus {modulus:?}")]
+    InvError { operand: BigInt, modulus: BigInt },
```

Comment:
Does `Inv` mean `Inverse`? Maybe it's better to use full meaning in this case, something like `NonexistantInverse`?

Reply:
ok